DataFormats/Detectors/EMCAL/src/Cell.cxx

```diff
@@ -23,176 +23,44 @@ Cell::Cell(Short_t tower, Double_t energy, Double_t time, ChannelType_t ctype)
+  return static_cast<Double_t>(timeInt);
 }
 
 void Cell::setEnergy(Double_t energy)
 {
-  ULong_t a = static_cast<ULong_t>(energy / 0.0153);
-  a = a & 0x3FFF;
-
-  a <<= 24;
-  ULong_t b = getLong() & 0xc000ffffff; // 1100000000000000111111111111111111111111
-  mBits = b + a;
+  double truncatedEnergy = energy;
+  if (truncatedEnergy < 0.) {
+    truncatedEnergy = 0.;
+  } else if (truncatedEnergy > 250.) {
+    truncatedEnergy = 250.;
+  }
+  mCellWord.mEnergy = static_cast<uint16_t>(truncatedEnergy / 0.0153);
 }
 
 Double_t Cell::getEnergy() const
 {
-  return double(getEnergyBits() * 0.0153);
-}
-
-void Cell::setType(ChannelType_t ctype)
-{
-  switch (ctype) {
-    case ChannelType_t::HIGH_GAIN:
-      setHighGain();
-      break;
-    case ChannelType_t::LOW_GAIN:
-      setLowGain();
-      break;
-    case ChannelType_t::TRU:
-      setTRU();
-      break;
-    case ChannelType_t::LEDMON:
-      setHighGain();
-      break;
-  };
-}
-
-ChannelType_t Cell::getType() const
-{
-  if (getHighGain())
-    return ChannelType_t::HIGH_GAIN;
-  else if (getLEDMon())
-    return ChannelType_t::LEDMON;
-  else if (getTRU())
-    return ChannelType_t::TRU;
-  return ChannelType_t::LOW_GAIN;
-}
-
-void Cell::setLowGain()
-{
-  std::bitset<40> b(0x3fffffffff); // 0011111111111111111111111111111111111111
-  mBits = (mBits & b);
-}
-
-Bool_t Cell::getLowGain() const
-{
-  ULong_t t = (getLong() >> 38);
-  if (t)
-    return false;
-  return true;
-}
-
-void Cell::setHighGain()
-{
-  ULong_t b = getLong() & 0x3fffffffff; // 0011111111111111111111111111111111111111
-  mBits = b + 0x4000000000;             // 0100000000000000000000000000000000000000
-}
-
-Bool_t Cell::getHighGain() const
-{
-  ULong_t t = (getLong() >> 38);
-  if (t == 1)
-    return true;
-  return false;
-}
-
-void Cell::setLEDMon()
-{
-  ULong_t b = getLong() & 0x3fffffffff; // 0011111111111111111111111111111111111111
-  mBits = b + 0x8000000000;             // 1000000000000000000000000000000000000000
-}
-
-Bool_t Cell::getLEDMon() const
-{
-  ULong_t t = (getLong() >> 38);
-  if (t == 2)
-    return true;
-  return false;
-}
-
-void Cell::setTRU()
-{
-  ULong_t b = getLong() & 0x3fffffffff; // 0011111111111111111111111111111111111111
-  mBits = b + 0xc000000000;             // 1100000000000000000000000000000000000000
-}
-
-Bool_t Cell::getTRU() const
-{
-  ULong_t t = (getLong() >> 38);
-  if (t == 3)
-    return true;
-  return false;
-}
-
-void Cell::setLong(ULong_t l)
-{
-  std::bitset<40> b(l);
-  mBits = b;
+  return static_cast<double>(mCellWord.mEnergy) * 0.0153;
 }
 
 void Cell::PrintStream(std::ostream& stream) const
 {
-  stream << "EMCAL Cell: Type " << getType() << ", Energy " << getEnergy() << ", Time " << getTimeStamp() << ", Tower " << getTower() << ", Bits " << mBits;
+  stream << "EMCAL Cell: Type " << getType() << ", Energy " << getEnergy() << ", Time " << getTimeStamp() << ", Tower " << getTower() << ", Bits " << std::bitset<40>(mCellWord.mDataWors);
```

Comment:
why this bitset is needed?

Reply:
Only for monitoring purpose, it is not really needed. I will remove the part.